src/App.tsx

```diff
@@ -1,93 +1,107 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState, useMemo, useContext } from 'react';
+import { Filter } from './types/Filter';
+import { TodosContext } from './components/TodosContext';
+import { Todolist } from './components/TodoList';
+import { TodoFilter } from './components/TodoFilter';
 
 export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [filter, setFilter] = useState<Filter>(Filter.All);
+
+  const todosContext = useContext(TodosContext);
+
+  const {
+    todos,
+    addTodo,
+    deleteCompletedTodos,
+    handleToggleAll,
+    todoCount,
+    completedTodos,
+    filterTodos,
+  } = todosContext;
```

Comment:
you can do it directly
```suggestion
  const {
    todos,
    addTodo,
    deleteCompletedTodos,
    handleToggleAll,
    todoCount,
    completedTodos,
    filterTodos,
  } = useContext(TodosContext);
```